sympy/functions/special/spherical_harmonics.py

```diff
@@ -178,7 +178,8 @@ def fdiff(self, argindex=4):
 
     def _eval_rewrite_as_polynomial(self, n, m, theta, phi, **kwargs):
         # TODO: Make sure n \in N
-        # TODO: Assert |m| <= n ortherwise we should return 0
+        if Abs(m) > n:  # Assert |m| <= n otherwise we should return 0
+            return 0
```

Comment:
What happens if `m` or `n` is a symbol?

Reply:
@oscarbenjamin I have also noticed this case, and I think that if m or n is a symbol, we can return a piecewise function that has two subfunctions, one subfunction for |m| > n and one subfunction for |m| < n. If you find this reasonable, I will update my pull request.